internal/controller/discoverymgr/mnedc/server/server.go

```diff
@@ -368,3 +371,23 @@ func (s *Server) Close() error {
 
 	return err
 }
+
+//GetVirtualIP returns the server virtualIP
+func (s *Server) GetVirtualIP() string {
+	return s.virtualIP.String()
+}
+
+//generateServerIP generates a virtual IP for the server
+func generateServerIP(privateIP string) {
```

Comment:
It might be good to return `net.IP` to allocate virtual IP to `Server.virtualIP`.
```suggestion
func generateServerIP() net.IP {
```